crypto/src/test/java/org/springframework/security/crypto/bcrypt/BCryptTests.java

```diff
@@ -14,18 +14,33 @@
 
 package org.springframework.security.crypto.bcrypt;
 
-import org.junit.Test;
+import static org.assertj.core.api.Assertions.assertThat;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
 
-import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.BeforeClass;
+import org.junit.Test;
 
 /**
  * JUnit unit tests for BCrypt routines
  * @author Damien Miller
  */
 public class BCryptTests {
 
+	private static class TestObject<T> {
```

Comment:
This class is a great addition to the test, it helps with readability and clarifies what the tests are doing.
Since it's not directly related to the `checkpw` addition, could you please move it to a separate PR or separate commit?

Reply:
Thanks for the feedback @eleftherias ! 
It does make sense splitting it into a separate PR and I have done so here: #7694 

Please note that to keep it as simplest as possible, this `TestObject` won't be using generics in that PR as the generic type would only be required when adding the ability to test a byte[] - which will be added as part of this change though!

I'll sort this branch to contain only the changes for `checkpw` once #7694 is approved and merged in 👍